devito/profiler.py

```diff
@@ -1,30 +1,86 @@
-from ctypes import Structure, byref, c_double
+from ctypes import Structure, byref, c_double, c_longlong
 
-from cgen_wrapper import Block, Statement, Struct, Value
+from cgen_wrapper import Assign, Block, For, Statement, Struct, Value
+from devito.logger import error
 
 
-class Profiler(Structure):
+class Profiler(object):
     """The Profiler class is used to manage profiling information for Devito
     generated C code.
     """
+    TIME = 1
+    FLOP = 2
 
-    def __init__(self):
-        self.name = "timings"
-        self.fields = []
-        self.timings = None
+    def __init__(self, openmp=False):
+        self.openmp = openmp
+        self.t_name = "timings"
+        self.o_name = "oi"
+        self.f_name = "flops"
+        self.loop_temp_prefix = "temp_"
+        self.t_fields = []
+        self.f_fields = []
+        self.oi = {}
+        self._timings = None
+        self._flops = None
+        self.flops_defaults = {}
 
-    def add_profiling(self, block, name, omp_flag=None):
+    def get_loop_reduction(self, op, variables):
+        """Function to generate reduction pragma, used for profiling under
+        OpenMP settings.
+        :param op: The reduction operator.
+        :param varaiables: A list of string, each is the name of a variable
+                           to be reduced.
+        :returns: String representing the reduction pragma
+        """
+        if len(variables) <= 0:
+            return ""
+        return " reduction(" + op + ":" + self.loop_temp_prefix + variables[0]\
+               + "".join(["," + self.loop_temp_prefix + v for v in variables[1:]]) + ")"
```

Comment:
Using `+` to concatenate strings is considered bad style by many. String formatting (`%s`) is much cleaner and more readable.